L1Trigger/L1THGCal/python/hgcalBackEndLayer2Producer_cfi.py

```diff
@@ -2,31 +2,110 @@
 
 import SimCalorimetry.HGCalSimProducers.hgcalDigitizer_cfi as digiparam
 import RecoLocalCalo.HGCalRecProducers.HGCalUncalibRecHit_cfi as recoparam
-import RecoLocalCalo.HGCalRecProducers.HGCalRecHit_cfi as recocalibparam 
-
-from L1Trigger.L1THGCal.egammaIdentification import egamma_identification_drnn_cone
-from L1Trigger.L1THGCal.customClustering import binSums, dr_layerbylayer
-
-C3d_parValues = cms.PSet( type_multicluster = cms.string('HistoMaxC3d'),
-                          dR_multicluster = cms.double(0.),
-                          dR_multicluster_byLayer_coefficientA = cms.vdouble(dr_layerbylayer),
-                          dR_multicluster_byLayer_coefficientB = cms.vdouble( [0]*53 ),
-                          minPt_multicluster = cms.double(0.5), # minimum pt of the multicluster (GeV)
-                          nBins_R_histo_multicluster = cms.uint32(36),
-                          nBins_Phi_histo_multicluster = cms.uint32(216),
-                          binSumsHisto = binSums,
-                          threshold_histo_multicluster = cms.double(10.),
-                          cluster_association = cms.string("NearestNeighbour"),
-                          EGIdentification=egamma_identification_drnn_cone.clone(),
-                          neighbour_weights=cms.vdouble(  0    , 0.25, 0   ,
-                                                          0.25 , 0  ,  0.25,
-                                                          0    , 0.25, 0
-                                                          )
- )
-
-be_proc = cms.PSet( ProcessorName  = cms.string('HGCalBackendLayer2Processor3DClustering'),
-                    C3d_parameters = C3d_parValues.clone()
-                  )
+import RecoLocalCalo.HGCalRecProducers.HGCalRecHit_cfi as recocalibparam
+
+from L1Trigger.L1THGCal.egammaIdentification import egamma_identification_drnn_cone, \
+                                                    egamma_identification_drnn_dbscan, \
+                                                    egamma_identification_histomax
+
+from Configuration.Eras.Modifier_phase2_hgcalV9_cff import phase2_hgcalV9
+
+
+binSums = cms.vuint32(13,               # 0
+                      11, 11, 11,       # 1 - 3
+                      9, 9, 9,          # 4 - 6
+                      7, 7, 7, 7, 7, 7,  # 7 - 12
+                      5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5,  # 13 - 27
+                      3, 3, 3, 3, 3, 3, 3, 3  # 28 - 35
+                      )
+
+EE_DR_GROUP = 7
+FH_DR_GROUP = 6
+BH_DR_GROUP = 12
+MAX_LAYERS = 52
```

Comment:
This is not correct for the V10 geometry. Will that cause any problems in the algorithms?

Reply:
The parameters for layers 51 and 52 will be discarded for the V10 geometry. The parameter vectors need to be as large or larger than the number of layers.